### PR TITLE
Feat/coinjoin status

### PR DESCRIPTION
**Pull request by szymonlesisz**
## Description

Based on https://github.com/trezor/trezor-suite/pull/6566

#### Add `Status` module to @trezor/coinjoin lib and implement it with `CoinjoinClient`

Once  `CoinjoinClient` is enabled it periodically sync with coordinator `/status` using random proxy (TOR) identities and time frequency until `client.disable();` is called.

> TOR identities works only in @trezor/suite-desktop, using suite-web build you can expect errors in console regarding `Proxy-Authorization` header which is used by @trezor/request-manager to establish TOR circuit
>
> Coinjoin is not meant to work in browser environment and this should be possible only in dev mode
>
> Sentry will catch that error if it happens in production

Status changes are emitted as `status` event:

```
client.on('status', event => {});

{
    rounds: Round[]; // current list of rounds
    changes: Round[]: // list of changed rounds since recent update
    feeRatesMedians: Array<{ timeFrame: string; medianFeeRate: number; }>, // timeFrame format: "0d 0h 0m 0s"
    coordinatorFeeRate: action.status.coordinatorFeeRate, // current coordinatorFeeRate
}
```

#### Handle status events in suite
Fields to use in coinjoin strategy components, discussed with @komret 
- feeRatesMedians.medianFeeRate values transformed from kvBytes to vBytes
  values from feeRatesMedians or from custom strategy **should be multiplied by 1000** before sent to Trezor (maxFeePerKvbyte param of authorizeCoinjoin action)
- coordinatorFeeRate value transformed to percentage
  coordinatorFeeRate **should be powered by 10** before sent to Trezor (maxCoordinatorFeeRate of authorizeCoinjoin sction)

### CODE REVIEW

**Comment by komret** [comment]
I would prefer to convert the `feeRatesMedians` on the reducer level to `maxMiningFee`
```ts
{
    recommended: number,
    fast: number,
}
```
The `fast` fee would also be multiplied by 2 (defined as a constant in config).

Possibly also recalculate the unit on `medianFeeRate` and `coordinatorFeeRate` here.

The goal is to have less calculations in the components and less data held in the application state.

---
BTW there is an error in the console when `/status` is called:
Screenshot 2022-10-14 at 15 24 46

**Comment by szymonlesisz** [comment]
> BTW there is an error in the console when /status is called

it's ok, it shoudlnt be working in browser, browser correctly blocks invalid header

this header is used to use custom tor identity (circuit) and its handled by ```suite-desktop > request-manager```

**Comment by szymonlesisz** [comment]
@komret pls checkout my fixup, now its like:
Screenshot from 2022-10-14 17-08-15

File: packages/suite/src/utils/wallet/coinjoinUtils.ts [hunk 1]
```diff
@@ -54,3 +55,28 @@ export const breakdownCoinjoinBalance = ({
...
     return balanceBreakdown;
 };
+
+/**
+ * Transform from coordinator format to coinjoinReducer format `CoinjoinClientFeeRatesMedians`
+ * array => object { name: value-in-vbytes }
+ */
+export const transformFeeRatesMedians = (m: CoinjoinStatusEvent['feeRatesMedians']) => {
+    const [fast, recommended, slow] = m.map(f => f.medianFeeRate);
+    const kb2b = (v: number) => (v ? Math.round(v / 1000) : 1);
```
Comment (komret) [inline packages/suite/src/utils/wallet/coinjoinUtils.ts]: The number is divided, so... is the original unit really kB?

**Comment by szymonlesisz** [comment]
also note that:
- values from feeRatesMedians or from custom strategy should be multiplied by 1000 before sent to Trezor (maxFeePerKvbyte)
- coordinatorFeeRate should be powered by 10 before sent to Trezor (maxCoordinatorFeeRate)

File: packages/suite/src/utils/wallet/coinjoinUtils.ts (hunk 1 above)
Comment (szymonlesisz) [reply packages/suite/src/utils/wallet/coinjoinUtils.ts]: accualy its kvbyte (kilo virtual byte) i'll rename it

Comment (szymonlesisz) [reply packages/suite/src/utils/wallet/coinjoinUtils.ts]: https://github.com/trezor/trezor-suite/pull/6563/commits/18004e06d620f2690e76a2416b43dca82270d791

**Comment by szymonlesisz** [comment]
reabsed over recent `feat/coinjoin-remember-forget` (recent develop)

**Comment by szymonlesisz** [comment]
squashed since there are no other change requests

File: packages/coinjoin/src/types/coordinator.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+export interface CoinjoinState {
+    Type: string; // TODO enum?
+    events: CoinjoinStateEvent[];
+    isFullySigned?: boolean;
+    witnesses?: Record<number, string>;
+}
+
+export enum RoundPhase {
+    InputRegistration = 0,
```
Comment (matejkriz) [inline packages/coinjoin/src/types/coordinator.ts:134]: nit: what is the point of setting the those initialization values explicitely? enums are numeric by default

**Review by matejkriz — COMMENTED** [review]
looks good to me, but please @dahaca do the review 👁️

**Review by dahaca — APPROVED** [review]

**Comment by komret** [comment]
@szymonlesisz I noticed that the coordinator fee rate is displayed in this modal before the value is fetched. It is hardcoded in config. So unless we want to keep two sources of truth, we should just use the value from config everywhere instead of the fetched `coordiantorFeeRate` value. Or not display the fee rate in the modal. What do you think?


Screenshot 2022-10-21 at 16 22 36

**Comment by komret** [comment]
> @szymonlesisz I noticed that the coordinator fee rate is displayed in this modal before the value is fetched. It is hardcoded in config. So unless we want to keep two sources of truth, we should just use the value from config everywhere instead of the fetched `coordiantorFeeRate` value. Or not display the fee rate in the modal. What do you think?

I hear this was already raised and product decision is needed. What do you think @hynek-jina or @sime? The simplest solution from the programming standpoint is not to show the concrete fee rate here, just something like "A coordinator fee rate is applied". But then we wouldn't display the fee rate anywhere, just the calculated fee.

**Comment by hynek-jina** [comment]
We can leave it hardcoded for now. And the fee is `up to 0.3%` (might be lower if you have some UTXO with amounts bellow 1M sats...)

**Pull request merged**